### PR TITLE
Update Python and related libraries to address CVE

### PR DESCRIPTION
**Pull request by rv0lt**
## Read this before submitting the PR

1. Always create a Draft PR first
2. Go through sections 1-5 below, fill them in and check all the boxes
3. Make sure that the branch is updated; if there's an "Update branch" button at the bottom of the PR, rebase or update branch.
4. When all boxes are checked, information is filled in, and the branch is updated: mark as Ready For Review and tag reviewers (top right)
5. Once there is a submitted review, implement the suggestions (if reasonable, otherwise discuss) and request an new review.

If there is a field which you are unsure about, enter the edit mode of this description or go to the [PR template](../.github/pull_request_template.md); There are invisible comments providing descriptions which may be of help.

## 1. Description / Summary

Update Python base image because of detected CVE
https://github.com/docker-library/python/issues/781

After inspecting the libraries, pyfakes also needs to be bumped because of dropped supported to the former version.
https://github.com/pytest-dev/pyfakefs/blob/main/CHANGES.md

Same with requests:
https://github.com/psf/requests/blob/main/HISTORY.md#2320-2024-05-20

wrapt:
https://wrapt.readthedocs.io/en/latest/changes.html#version-1-14-0

## 2. Jira task / GitHub issue

[_Link to the github issue or add the Jira task ID here._](https://scilifelab.atlassian.net/jira/software/projects/DDS/boards/13?selectedIssue=DDS-2038)

## 3. Type of change

What _type of change(s)_ does the PR contain?

**Check the relevant boxes below. For an explanation of the different sections, enter edit mode of this PR description template.**

- [ ] New feature
  - [ ] Breaking: _Why / How? Add info here._ 
  - [ ] Non-breaking 
- [ ] Database change: _Remember the to include a new migration version, **or** explain here why it's not needed._ 
- [ ] Bug fix 
- [X] Security Alert fix 
  - [X] Package update 
    - [X] Major version update 
- [ ] Documentation 
- [ ] Workflow 
- [ ] Tests **only** 

## 4. Additional information

- [X] [Sprintlog](../SPRINTLOG.md) 
- [ ] Blocking PRs 
  - [ ] Merged 
- [ ] PR to `master` branch: \_If checked, read [the release instructions](../doc/procedures/new_release.md) 
  - [ ] I have followed steps 1-8. 

## 5. Actions / Scans

_Check the boxes when the specified checks have passed._

**For information on what the different checks do and how to fix it if they're failing, enter edit mode of this description or go to the [PR template](../.github/pull_request_template.md).**

- [X] **Black**

- [X] **Prettier**

- [X] **Yamllint**

- [X] **Tests**

- [X] **CodeQL**

- [X] **Trivy**

- [X] **Snyk**

### CODE REVIEW

File: Dockerfiles/backend.Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #############################
 
 # Set official image -- parent image
-FROM python:3.11-alpine as base
+FROM python:3.12-alpine as base
```
Comment (aishling-scilifelab) [inline Dockerfiles/backend.Dockerfile:6]: Do we want to update some corresponding Python stuff in dds_cli? 

https://github.com/ScilifelabDataCentre/dds_cli/blob/7fefe4c083883f544f1b5a32d34b1ea98de89ff2/setup.py#L27 

https://github.com/ScilifelabDataCentre/dds_cli/blob/7fefe4c083883f544f1b5a32d34b1ea98de89ff2/WINDOWS.md?plain=1#L3

Comment (rv0lt) [reply Dockerfiles/backend.Dockerfile:6]: That would need to go in a different PR anyway. Maybe better to comment it on the jira task?

Comment (aishling-scilifelab) [reply Dockerfiles/backend.Dockerfile:6]: Yup, more just a thought that this should be part of this ticket (but not this PR).

File: SPRINTLOG.md [hunk 2]
```diff
@@ -425,3 +425,4 @@ _Nothing merged during this sprint_
 
 - Update certifi to remove GLOBALISSUER certicates ([#1549](https://github.com/ScilifelabDataCentre/dds_web/pull/1549))
 - Add CODEOWNERS file in order to define Team Hermes as owners of all files in repository ([#708](https://github.com/ScilifelabDataCentre/dds_web/pull/708))
+- Update python base image and related libraries ([#1548](https://github.com/ScilifelabDataCentre/dds_web/pull/1548))
```
Comment (i-oden) [inline SPRINTLOG.md:428]: This is a minor thing but I'd like to have this info in the sprintlog 
```suggestion
- Bump python base image to 3.12 and related libraries ([#1548](https://github.com/ScilifelabDataCentre/dds_web/pull/1548))
```

**Review by i-oden — CHANGES_REQUESTED** [review]
I'm not able to start up the containers locally, I believe it's due to using eduroam since I still don't have an ethernet cable. So I can't test this unfortunately atm.

**Comment by i-oden** [comment]
Have marked as request changes since I'd like to be able to test this prior to merging.

**Review by valyo — APPROVED** [review]
I started the local environment and tested manually the following:
backend container: python version 3.12.5, requests version 2.32.0
client container: python version 3.11.9
login with all different roles, cli and web
listing projects cli and web
creation of project
upload
releasing of project
inviting users and account registration
adding researcher to project
download
various superadmin commands: maintenance, motd, user delete

I didn't got any errors and didn't notice any strange behaviour

**Review by valyo — CHANGES_REQUESTED** [review]
The local dev environment doesn't start for me, and throws an error that seems to be related to the package `wrapt`, which seems to use the module `inspect`, which is no longer available in Python 3.12. That's how far I went checking it.
The strange thing is that this issue didn't show up before adding the cli python version bump

**Comment by valyo** [comment]
`said library` ?! maybe autocorrection
It is strange that 1) you don't get errors in your containers; and 2) that I didn't got the either when I tested before.

**Comment by rv0lt** [comment]
> `said library` ?! maybe autocorrection It is strange that 1) you don't get errors in your containers; and 2) that I didn't got the either when I tested before.

Which command are you using to spin off the containers? _docker-compse up --profile cli_ ?

**Comment by valyo** [comment]
> Which command are you using to spin off the containers? _docker-compse up --profile cli_ ?

yes

**Review by valyo — APPROVED** [review]
Works fine now. 
I haven't tested as extensively as before, just login, project creation, upload and download

**Review by i-oden — CHANGES_REQUESTED** [review]
Looks good, just waiting for CLI to also be approved and then I'll approve this one.

**Review by i-oden — APPROVED** [review]

**Pull request merged**